Buildings/Controls/OBC/ASHRAE/PrimarySystem/ChillerPlant/MinimumFlowBypass/Controller.mo

```diff
@@ -0,0 +1,223 @@
+      iconTransformation(extent={{-120,80},{-100,100}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput VChiWat_flow(
+    final min=0,
+    final unit="m3/s",
+    quantity="VolumeFlowRate")
+    "Measured chilled water flow rate through chillers"
+    annotation (Placement(transformation(extent={{-140,70},{-100,110}}),
+      iconTransformation(extent={{-120,60},{-100,80}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uStaUp "Stage up logical signal"
+    annotation (Placement(transformation(extent={{-140,40},{-100,80}}),
+      iconTransformation(extent={{-120,40},{-100,60}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uUpsDevSta
+    "Status of resetting status of device before reset minimum bypass flow setpoint"
+    annotation (Placement(transformation(extent={{-140,10},{-100,50}}),
+      iconTransformation(extent={{-120,20},{-100,40}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uEnaNexChi
+    "Status to indicate that it starts to enable another chiller. This input used when the stage change needs chiller on/off"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+      iconTransformation(extent={{-120,0},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uOnOff
+    "Indicate if the stage require one chiller to be enabled while another is disabled"
+    annotation (Placement(transformation(extent={{-140,-50},{-100,-10}}),
+      iconTransformation(extent={{-120,-20},{-100,0}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uChi[nChi]
+    "Chiller status: true=ON"
+    annotation (Placement(transformation(extent={{-140,-80},{-100,-40}}),
+      iconTransformation(extent={{-120,-40},{-100,-20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput nexEnaChi
+    "Index of next enabling chiller"
+    annotation (Placement(transformation(extent={{-140,-110},{-100,-70}}),
+      iconTransformation(extent={{-120,-60},{-100,-40}})));
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerInput nexDisChi
+    "Index of next disabling chiller"
+    annotation (Placement(transformation(extent={{-140,-140},{-100,-100}}),
+      iconTransformation(extent={{-120,-80},{-100,-60}})));
+  Buildings.Controls.OBC.CDL.Interfaces.BooleanInput uStaDow
+    "Indicate if there is stage down"
+    annotation (Placement(transformation(extent={{-140,-170},{-100,-130}}),
+      iconTransformation(extent={{-120,-100},{-100,-80}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput yValPos(
+    final min=0,
+    final max=1,
+    final unit="1") "Chilled water minimum flow bypass valve position"
+    annotation (Placement(transformation(extent={{100,120},{120,140}}),
+      iconTransformation(extent={{100,-10},{120,10}})));
+
+  Buildings.Controls.OBC.CDL.Continuous.LimPID valPos(
+    final controllerType=controllerType,
+    final k=k,
+    final Ti=Ti,
+    final Td=Td,
+    final yMax=yMax,
+    final yMin=yMin,
+    final reverseAction=true,
```

Comment:
This controller should be in direct action right?
When `V < VSet` i.e. `trkErr = VSet - V > 0` we want to open the bypass valve so `y > 0` thus `K > 0`.